core/src/parking_lot.rs

```diff
@@ -16,6 +16,39 @@ use core::{
 use smallvec::SmallVec;
 use std::time::{Duration, Instant};
 
+cfg_if! {
+    if #[cfg(target_arch = "wasm32-unknown-unknown")] {
+        use core::ops::Add;
+
+        #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+        struct DummyInstant(Duration);
+
+        impl DummyInstant {
+            pub fn now() -> DummyInstant {
+                DummyInstant::zero()
+            }
+
+            const fn zero() -> DummyInstant {
+                DummyInstant(Duration::from_secs(0))
+            }
+        }
+
+        impl Add<Duration> for DummyInstant {
+            type Output = DummyInstant;
+
+            fn add(self, _rhs: Duration) -> DummyInstant {
+                DummyInstant::zero()
+            }
+        }
+
+        // Use dummy implementation for `Instant` on `wasm32`
```

Comment:
Add a small comment explaining that `Instant::now()` always panics on wasm32-unknown-unknown.